File: App Mobile/flutter_store_shoe/lib/models/ProductInfoVM.dart
// ignore_for_file: file_names

import 'package:flutter_store_shoe/models/RatingModel.dart';

class ProductDetail {
  int id;
  int productId;
  double price;
  String? codeColor;
  String? urlImage;

  ProductDetail({
    required this.id,
    required this.productId,
    required this.price,
    this.codeColor,
    this.urlImage,
  });

  factory ProductDetail.fromJson(Map<String, dynamic> json) {
    return ProductDetail(
      id: json['id'] ?? '',
      productId: json['productId'],
      price: json['price'].toDouble(),
      codeColor: json['codeColor'],
      urlImage: json['urlImage'],
    );
  }
}

class Product {
  int id;
  String productName;
  String productDescription;
  List<ProductDetail>? details;
  bool isFav = false;
  RatingModel? rating;
  Product({
    required this.id,
    required this.productName,
    required this.productDescription,
    this.details,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    var detailsList = json['details'] as List;
    List<ProductDetail> details =
        detailsList.map((detail) => ProductDetail.fromJson(detail)).toList();
    return Product(
      id: json['id'] ?? 0,
      productName: json['productName'] ?? '',
      productDescription: json['prodductDescription'] ?? '',
      details: details,
    );
  }
}

File: Store-Shoe-Online/Store-Shoe-Online/ViewModel/ProductInfoVM.cs
using System.ComponentModel.DataAnnotations;

namespace Store_Shoe_Online.ViewModel
{
    public class ProductInfoVM
    {
        public int Id { get; set; }
        public string ProductName { get; set; }
        public string ProdductDescription { get; set; }
        public int CategoryId { get; set; }
        [DisplayFormat(DataFormatString = "{0:0.###}")]
        public double Rating { get; set; }
        public bool IsFavorite { get; set; }
        // Các thuộc tính khác bạn muốn bao gồm
        public List<ProductDetailDTO> Details { get; set; }
    }
    public class ProductDetailDTO
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public double Price { get; set; }
        public string CodeColor { get; set; }
        public string urlImage { get; set; }
    }
}
